internal/configsource/source_test.go

```diff
@@ -367,46 +367,6 @@ func TestConfigSourceManagerMultipleWatchForUpdate(t *testing.T) {
 	assert.NoError(t, callClose(closeFunc))
 }
 
-func TestConfigSourceManagerEnvVarHandling(t *testing.T) {
-	require.NoError(t, os.Setenv("envvar", "envvar_value"))
-	defer func() {
-		assert.NoError(t, os.Unsetenv("envvar"))
-	}()
-
-	tstCfgSrc := TestConfigSource{
-		ValueMap: map[string]valueEntry{
-			"int_key": {Value: 42},
-		},
-	}
-
-	// Intercept "params_key" and create an entry with the params themselves.
-	tstCfgSrc.OnRetrieve = func(_ context.Context, selector string, paramsConfigMap *confmap.Conf) error {
-		var val any
-		if paramsConfigMap != nil {
-			val = paramsConfigMap.ToStringMap()
-		}
-		if selector == "params_key" {
-			tstCfgSrc.ValueMap[selector] = valueEntry{Value: val}
-		}
-		return nil
-	}
-
-	file := path.Join("testdata", "envvar_cfgsrc_mix.yaml")
-	cp, err := confmaptest.LoadConf(file)
-	require.NoError(t, err)
-
-	expectedFile := path.Join("testdata", "envvar_cfgsrc_mix_expected.yaml")
-	expectedParser, err := confmaptest.LoadConf(expectedFile)
-	require.NoError(t, err)
-
-	res, closeFunc, err := ResolveWithConfigSources(context.Background(), map[string]ConfigSource{"tstcfgsrc": &tstCfgSrc}, nil, cp, func(_ *confmap.ChangeEvent) {
-		panic("must not be called")
-	})
-	require.NoError(t, err)
-	assert.Equal(t, expectedParser.ToStringMap(), res.ToStringMap())
-	assert.NoError(t, callClose(closeFunc))
-}
-
```

Comment:
The test has been removed since the expansion was disabled in favor of the upstream implementation. e2e functionality is covered in the integration tests